include/network_interface/network_utils.h

```diff
@@ -8,21 +8,19 @@
 #ifndef NETWORK_INTERFACE_NETWORK_UTILS_H
 #define NETWORK_INTERFACE_NETWORK_UTILS_H
 
-#include <cstddef>
+#include "network_interface/common.h"
+
 #include <cstdint>
 #include <vector>
 #include <typeinfo>
+#include <string>
 #include <cstring>
+#include <type_traits>
 
 namespace AS
 {
 namespace Network
 {
-enum ByteOrder
-{
-  BE = 0,
-  LE
-};
 
 inline bool system_is_big_endian()
 {
```

Comment:
As I found out the other day, this is relying on undefined behavior in C++

Reply:
You're correct. In order for the compiler not to balk at it, I had to do a `reinterpret_cast` on a pointer to the type but it does work correctly (as shown by the tests). Should be good now.

Reply:
All the tests pass, but the CircleCI machines run x86_64 same as our dev machines. Since x64 is what this is most likely to be run on that's probably good. However a lot of ARM machines run big-endian so I'd like to test on a BE system if we plan on supporting any other architectures.

Reply:
There is a problem running non-x86 images on CircleCI. See https://support.circleci.com/hc/en-us/articles/360018651374-Job-completes-after-Spin-up-Environment-for-non-x86-based-Docker-images-. I tested this and they fail exactly as described. I've asked for more information at https://ideas.circleci.com/ideas/CCI-I-900 but I doubt I'll see a fix anytime soon. I'll test this locally to see if I can get it to run.

Reply:
@JWhitleyAStuff Did you have any success with testing locally?

Reply:
I haven't had time, so no. Also, I couldn't get the stock cross-platform `ros` Docker images to run on my machine (even after doing the qemu instructions to get them going) though I could run the cross-platform `ubuntu` images.

Reply:
Have you since had time to test on a local machine?

Reply:
@srustanAStuff - I finally just got it installed and running. The tests passed correctly. I'm going to try to set up CI to do this test as well. Please hold on approval until I have that added to this PR.

Reply:
Holy carp, it worked! Now testing (and passing) against `armhf` on CircleCI.